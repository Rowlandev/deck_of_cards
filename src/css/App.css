html, body, #root {
  height: 100%;
  margin: 0px;
  padding: 0px;
}

.App {
  background: #76b852; /* fallback for old browsers */
  background: -webkit-linear-gradient(to right, #76b852, #8dc26f); /* Chrome 10-25, Safari 5.1-6 */
  background: linear-gradient(to right, #76b852, #8dc26f); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
  height: 100%;
}

#top-left-suite {
  margin: 0px;
  pading: 0px;
  background: red;
  width: auto;
}

#top-left-value {
  margin: 0px;
  pading: 0px;
  background: blue;
}

#bottom-right-suite {
  margin: 0px;
  pading: 0px;
  background: yellow;
}

#bottom-right-value {
  margin: 0px;
  pading: 0px;
  background: purple;
}
