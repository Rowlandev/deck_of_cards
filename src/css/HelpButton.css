.help-icon {
  text-align: center;
}

#help-icon {
  width: 50px;
  height: 50px;
  margin-top: 1%;
  border-radius: 100%;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
  transition: 0.3s;
}

#help-icon:hover {
  cursor: pointer;
  filter: brightness(90%);
  transition: 0.3s;
}
