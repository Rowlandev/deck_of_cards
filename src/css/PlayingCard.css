#playing-card {
  background: white;
  width: 143px;
  height: 200px;
  border-radius: 7%;
  margin: 0 auto;
  overflow: hidden;
  position: relative; /* prevents child elements from going outside of itself */
  font-family: 'Oswald', sans-serif;
  top: 100px;
}

#top-left-suite {
  position: absolute;
	top: 5%;
	left: 5%;
}

#top-left-value {
  position: absolute;
  top: 7%;
  left: 11.5%;
  font-weight: bold;
  font-size: 1.3em;
}

#bottom-right-suite {
  position: absolute;
	bottom: 5%;
	right: 5%;
}

#bottom-right-value {
  position: absolute;
  bottom: 7%;
  right: 11.5%;
  font-weight: bold;
  font-size: 1.3em;
}

#large-suite {
  position: absolute;
  top: 31%;
  left: 23%;
}

.red {
  color: red;
}

.black {
  color: black;
}
