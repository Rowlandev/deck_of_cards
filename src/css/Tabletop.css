.main-menu {
  text-align: center;
  margin: 0;
}

#title {
  margin: 0;
  text-align: center;
  font-family: "Andale Mono", AndaleMono, monospace;
  font-size: 75px;
  color: white;
  padding-top: 40px;
}

#creators {
  font-family: "Andale Mono", AndaleMono, monospace;
  font-size: 20px;
  color: white;
  padding-bottom: 15px;
}

.button {
  background-color: white;
  width: 250px;
  border: solid black 1px;
  color: black;
  padding: 20px;
  font-family: "Andale Mono", AndaleMono, monospace;
  text-align: center;
  display: inline-block;
  font-size: 16px;
  border-radius: 7px;
  margin-top: 7px;
}

.button:hover {
  background-color: LightGray;
}

#settings-menu {
  text-align: center;
  margin: 0;
}

#settings-title {
  margin: 0;
  text-align: center;
  font-family: "Andale Mono", AndaleMono, monospace;
  font-size: 50px;
  color: white;
  padding-top: 40px;
  padding-bottom: 40px;
}

#card-choices {
  border: solid white 1px;
  border-radius: 7px;
  background: black;
  margin: auto;
  width: 650px;
}

.card-backs {
  width: 143px;
  height: 200px;
  padding: 0;
  border-radius: 7%;
  border: solid gray 1px;
  margin: 0 auto;
  margin-left: 5px;
  margin-right: 5px;
  overflow: hidden;
}
