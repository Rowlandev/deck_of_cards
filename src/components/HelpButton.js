import React from "react";
import HelpIcon from "../images/icons/passive-help.png";
import "../css/HelpButton.css";

class HelpButton extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      showing: this.props.showing,
      modalClassName: "modal-hidden"
    };
  }

  toggleHelp = () => {

    var classNameToBe;

    if (this.state.modalClassName === "modal-hidden") {
      classNameToBe = "modal-showing";
    } else {
      classNameToBe = "modal-hidden";
    }

    this.setState({
      showing: !this.state.showing,
      modalClassName: classNameToBe
    });
  }

  render() {
    return (
      <div className="help-modal-container">

        {/* Darkened Background*/}
        <div className={this.state.modalClassName}></div>

        {/* Help Icon */}
        <img src={HelpIcon} alt="help icon" id="help-icon" onClick={this.toggleHelp}/>

        {/* Header & Body*/}
        {this.state.showing &&
          <div id="header-and-body">

            <div id="header">
              <p id='header-text'>Welcome to our deck of cards simulation!</p>
            </div>

            <div id="body">
              <p id='body-text'>
                Using our deck of cards simulation is very
                simple - all you have to do is drag and drop cards
                and pretend as if they were real :)
              </p>
            </div>

            <div id="close-button" onClick={this.toggleHelp}>
              Close
            </div>

          </div>
        }

        {/* Close Modal Button */}


      </div>
    );
  }
}

export default HelpButton;
