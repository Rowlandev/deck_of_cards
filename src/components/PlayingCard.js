import React from "react";
import "../css/PlayingCard.css";

// import red corner suit icons
import RedDiamondCorner from "../images/red-corners/diamond.png";
import RedHeartCorner from "../images/red-corners/heart.png";

// import black corner suit icons
import BlackClubCorner from "../images/black-corners/club.png";
import BlackSpadeCorner from "../images/black-corners/spade.png";

// import red large icons
import RedDiamondLarge from "../images/red-large/diamond.png";
import RedHeartLarge from "../images/red-large/heart.png";

// import black large icons
import BlackClubLarge from "../images/black-large/club.png";
import BlackSpadeLarge from "../images/black-large/spade.png";

// import GuyFiere from "../images/backs/guy-fieri.jpg"
import Prof from "../images/backs/prof.jpg"

class PlayingCard extends React.Component{

  constructor(props) {
    super(props);
    this.state = {
      suit: this.props.suit,
      value: this.props.value,
      top: 50,
      left: 50,
      mouseDownX:0,
      mouseDownY:0,
      side: "back",
      clicked: false
    };
  }

  cardClicked = (e) => {
    this.mouseDownX = e.clientX;
    this.mouseDownY = e.clientY;
    this.setState({
      clicked: true
    })
  }

  cardReleased = (e) => {
    var newSide = this.state.side;  //in case it is not changed
    var newTop = this.state.top;
    var newLeft = this.state.left;
    var x = e.clientX;
    var y = e.clientY;


    if (this.state.clicked === true){
      if (x === this.mouseDownX && y === this.mouseDownY) {
        if (this.state.side === "back") {
          newSide = "front";
        } else {
          newSide = "back";
        }
      } else {
        newTop = this.state.top + (y - this.mouseDownY);
        newLeft = this.state.left + (x - this.mouseDownX);
      }
    }

    this.setState({
      suit: this.props.suit,
      value: this.props.value,
      top: newTop,
      left: newLeft,
      side: newSide,
      clicked: false
    });
  }

  render() {

    // define variables for card values
    var topLeftSuit;
    var topLeftValue;
    var bottomRightSuit;
    var bottomRightValue;
    var largeSuit;
    var offset = {top: this.state.top + "px",
                  left: this.state.left + "px"
                }

    // assign card red suit

    if (this.state.suit === "heart") {
      topLeftSuit = <img src={RedHeartCorner} alt="suit of hearts" id="top-left-suit"/>;
      bottomRightSuit = <img src={RedHeartCorner} alt="suit of hearts" id="bottom-right-suit"/>;
      largeSuit = <img src={RedHeartLarge} alt="large heart" id="large-suit"/>
    } else if (this.state.suit === "diamond") {
      topLeftSuit = <img src={RedDiamondCorner} alt="suit of diamonds" id="top-left-suit"/>;
      bottomRightSuit = <img src={RedDiamondCorner} alt="suit of diamonds" id="bottom-right-suit"/>;
      largeSuit = <img src={RedDiamondLarge} alt="large heart" id="large-suit"/>
    } else if (this.state.suit === "spade") {
        topLeftSuit = <img src={BlackSpadeCorner} alt="suit of spades" id="top-left-suit"/>;
        bottomRightSuit = <img src={BlackSpadeCorner} alt="suit of spades" id="bottom-right-suit"/>;
        largeSuit = <img src={BlackSpadeLarge} alt="large heart" id="large-suit"/>
    } else if (this.state.suit === "club") {
      topLeftSuit = <img src={BlackClubCorner} alt="suit of clubs" id="top-left-suit"/>;
      bottomRightSuit = <img src={BlackClubCorner} alt="suit of clubs" id="bottom-right-suit"/>;
      largeSuit = <img src={BlackClubLarge} alt="large heart" id="large-suit"/>
    }

    // assign card value based on props
    switch (this.state.value) {
      case "A":
        topLeftValue = <p className={this.state.suit} id="top-left-value">A</p>;
        bottomRightValue = <p className={this.state.suit} id="bottom-right-value">A</p>;
        break;
      case "2":
        topLeftValue = <p className={this.state.suit} id="top-left-value">2</p>;
        bottomRightValue = <p className={this.state.suit} id="bottom-right-value">2</p>;
        break;
      case "3":
        topLeftValue = <p className={this.state.suit} id="top-left-value">3</p>;
        bottomRightValue = <p className={this.state.suit} id="bottom-right-value">3</p>;
        break;
      case "4":
        topLeftValue = <p className={this.state.suit} id="top-left-value">4</p>;
        bottomRightValue = <p className={this.state.suit} id="bottom-right-value">4</p>;
        break;
      case "5":
        topLeftValue = <p className={this.state.suit} id="top-left-value">5</p>;
        bottomRightValue = <p className={this.state.suit} id="bottom-right-value">5</p>;
        break;
      case "6":
        topLeftValue = <p className={this.state.suit} id="top-left-value">6</p>;
        bottomRightValue = <p className={this.state.suit} id="bottom-right-value">6</p>;
        break;
      case "7":
        topLeftValue = <p className={this.state.suit} id="top-left-value">7</p>;
        bottomRightValue = <p className={this.state.suit} id="bottom-right-value">7</p>;
        break;
      case "8":
        topLeftValue = <p className={this.state.suit} id="top-left-value">8</p>;
        bottomRightValue = <p className={this.state.suit} id="bottom-right-value">8</p>;
        break;
      case "9":
        topLeftValue = <p className={this.state.suit} id="top-left-value">9</p>;
        bottomRightValue = <p className={this.state.suit} id="bottom-right-value">9</p>;
        break;
      case "10":
        topLeftValue = <p className={this.state.suit} id="top-left-value">10</p>;
        bottomRightValue = <p className={this.state.suit} id="bottom-right-value">10</p>;
        break;
      case "Jack":
        topLeftValue = <p className={this.state.suit} id="top-left-value">J</p>;
        bottomRightValue = <p className={this.state.suit} id="bottom-right-value">J</p>;
        break;
      case "Queen":
        topLeftValue = <p className={this.state.suit} id="top-left-value">Q</p>;
        bottomRightValue = <p className={this.state.suit} id="bottom-right-value">Q</p>;
        break;
      case "King":
        topLeftValue = <p className={this.state.suit} id="top-left-value">K</p>;
        bottomRightValue = <p className={this.state.suit} id="bottom-right-value">K</p>;
        break;
      case "Joker":
        topLeftValue = <p className={this.state.suit} id="top-left-value">JOKER</p>;
        bottomRightValue = <p className={this.state.suit} id="bottom-right-value">JOKER</p>;
        break;
      default:
        break;
    }





    return (
      <div>
      {this.state.side === "back" &&
        <div id="back-of-card" style={offset} data-testid="card" onMouseDown={(e) => this.cardClicked(e)} onMouseUp={(e) => this.cardReleased(e)}>
          <img src={Prof} alt="guy fieri" id="back-image"/>
        </div>
      }

      {this.state.side === "front" &&
        <div id="playing-card" style={offset} data-testid="card" onMouseDown={(e) => this.cardClicked(e)} onMouseUp={(e) => this.cardReleased(e)}>
          {topLeftSuit}
          {bottomRightSuit}
          {topLeftValue}
          {bottomRightValue}
          {largeSuit}
        </div>
      }
      </div>
    );
  }
}

export default PlayingCard;
