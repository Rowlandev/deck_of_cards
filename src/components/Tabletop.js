import React from "react";
import PlayingCard from "./PlayingCard.js";
import HelpButton from "./HelpButton.js";

class Tabletop extends React.Component {
  render() {
    return (
      <div>
        <HelpButton/>
        <PlayingCard suite="heart" value="5" color="red"/>
      </div>
    );
  }
}

export default Tabletop;
