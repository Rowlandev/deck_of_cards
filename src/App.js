import React from 'react';
import Tabletop from "./components/Tabletop.js";
import "./css/App.css";

function App() {
  return <div className="App"><Tabletop/></div>;
}

export default App;
