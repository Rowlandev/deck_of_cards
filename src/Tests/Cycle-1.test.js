// SYSTEM IMPORTS
import React from "react";
import ReactDOM from "react-dom";

// TESTING TOOL IMPORTS
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import '@testing-library/jest-dom/extend-expect';
import { render } from "@testing-library/react";

// COMPONENTS TO BE TESTED
import App from "../App.jsu";
import HelpButton from "../components/HelpButton.js";
import PlayingCard from "../components/PlayingCard.js";
import Tabletop from "../components/Tabletop.js";

/* CONFIGURE ENZYME */
Enzyme.configure({ adapter: new Adapter() });

/* SET OF TESTS FOR FIRST DEVELOPMENT CYCLE */
describe('CYCLE 1 TESTS', () => {

  var helpButtonWrapper = shallow(<HelpButton showing={false}/>); // get App component
  var playingCardWrapper = shallow(<PlayingCard suite="heart" value="5" color="red"/>); // get SidePanel component
  var tabletopWrapper = shallow(<Tabletop/>);

  var helpButtonInstacne = helpButtonWrapper.instance();
  var playingCardInstance = playingCardWrapper.instance();
  var tabletopInstance = tabletopWrapper.instance();

  /* TEST #1 - COMPONENTS RENDER CORRECTLY */
  it('COMPONENTS RENDER CORRECTLY', () => {
    const { asFragment } = render(<App/>); // GENERATES SNAPSHOT OF TABLETOP
    expect(asFragment()).toMatchSnapshot(); // CHECKS IF SNAPSHOT MATCHES RECENTLY RENDERED COMPONENT
  })

});
