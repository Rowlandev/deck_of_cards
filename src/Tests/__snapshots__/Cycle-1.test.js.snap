// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CYCLE 1 TESTS COMPONENTS RENDER CORRECTLY 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <div
        class="help-icon"
      >
        <img
          alt="help icon"
          id="help-icon"
          src="passive-help.png"
        />
      </div>
      <div
        id="back-of-card"
      >
        <div
          data-testid="card"
          id="playing-card"
        >
          <img
            alt="guy fieri"
            id="back-image"
            src="prof.jpeg"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
